analysis/deathknightblood/src/modules/features/BoneShieldTimesByStacks.js

```diff
@@ -44,9 +64,35 @@ class BoneShieldStacksBySeconds extends Analyzer {
       return;
     }
     this.lastBoneShieldUpdate = event.timestamp;
+    const nextStacks = currentStacks(event);
+    const didExpire = nextStacks === 0? this.didExpire(event) : false;
+    if (nextStacks < this.lastBoneShieldStack && !didExpire) {
+      // TODO: Blood Tap cdr
+      this.reduceDRWCooldown(nextStacks - this.lastBoneShieldStack);
+    }
     this.lastBoneShieldStack = currentStacks(event);
   }
 
+  trackApplication(event) {
+    // Can't track this with buff applied events in case the player
+    // refreshes at 10 stacks.
+    const isMarrow = (event.ability.guid === SPELLS.MARROWREND.spellID);
+    const isDRW = (event.ability.guid === SPELLS.DANCING_RUNE_WEAPON.spellID);
+    const hasCRW = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.CRIMSON_RUNE_WEAPON.bonusID);
```

Comment:
just merged something into the main `shadowlands` branch that changes how legendary checks worked. you'll want to pull that into your branch and change this to

```suggestion
    const hasCRW = this.selectedCombatant.hasLegendary(SPELLS.CRIMSON_RUNE_WEAPON);
```